api/ccapi/src/factory.cpp

```diff
@@ -40,6 +40,161 @@ std::unique_ptr<Layer> createLayer(LayerType type,
+}
+
+/**
+ * @brief Helper function to create convolution 2d layer
+ */
+std::unique_ptr<Layer>
+Convolution2D(const std::vector<std::string> &properties) {
+  return createLayer(LayerType::LAYER_CONV2D, properties);
+}
+
+/**
+ * @brief Helper function to create pooling 2d layer
+ */
+std::unique_ptr<Layer> Pooling2D(const std::vector<std::string> &properties) {
+  return createLayer(LayerType::LAYER_POOLING2D, properties);
+}
+
+/**
+ * @brief Helper function to create flatten layer
+ */
+std::unique_ptr<Layer> Flatten(const std::vector<std::string> &properties) {
+  return createLayer(LayerType::LAYER_FLATTEN, properties);
+}
+
+/**
+ * @brief Helper function to create activation layer
+ */
+static std::unique_ptr<Layer>
+Activation(const std::vector<std::string> &properties) {
+  return createLayer(LayerType::LAYER_ACTIVATION, properties);
+}
+
+/**
+ * @brief Helper function to create ReLU activation layer
+ */
+std::unique_ptr<Layer> ReLU(const std::vector<std::string> &properties) {
+  std::vector<std::string> props(properties);
+  props.push_back("Activation=relu");
+  return Activation(props);
+}
+
+/**
+ * @brief Helper function to create Tanh layer
+ */
+std::unique_ptr<Layer> Tanh(const std::vector<std::string> &properties) {
+  std::vector<std::string> props(properties);
+  props.push_back("Activation=tanh");
+  return Activation(props);
+}
+
+/**
+ * @brief Helper function to create sigmoid layer
+ */
+std::unique_ptr<Layer> Sigmoid(const std::vector<std::string> &properties) {
+  std::vector<std::string> props(properties);
+  props.push_back("Activation=sigmoid");
+  return Activation(props);
+}
+
+/**
+ * @brief Helper function to create softmax layer
+ */
+std::unique_ptr<Layer> Softmax(const std::vector<std::string> &properties) {
+  std::vector<std::string> props(properties);
+  props.push_back("Activation=softmax");
+  return Activation(props);
+}
+
+/**
+ * @brief Helper function to create addition layer
+ */
+std::unique_ptr<Layer> Addition(const std::vector<std::string> &properties) {
+  return createLayer(LayerType::LAYER_ADDITION, properties);
+}
+
+/**
+ * @brief Helper function to create concat layer
+ */
+std::unique_ptr<Layer> Concat(const std::vector<std::string> &properties) {
+  return createLayer(LayerType::LAYER_CONCAT, properties);
+}
+
+/**
+ * @brief Helper function to create multi-out layer
+ */
+std::unique_ptr<Layer> MultiOut(const std::vector<std::string> &properties) {
+  return createLayer(LayerType::LAYER_OUT, properties);
+}
+
+} // namespace layer
+
+/**
+ * @brief Factory creator with constructor for layer
+ */
+static std::unique_ptr<Layer>
+createLoss(nntrainer::LossType type,
+           const std::vector<std::string> &properties) {
+  std::unique_ptr<Layer> layer = nntrainer::createLoss(type);
+
+  if (layer->setProperty(properties) != ML_ERROR_NONE)
+    throw std::invalid_argument("Set properties failed for layer");
+
+  return layer;
+}
+
+namespace loss {
```

Comment:
Are we gonna open up loss to be generated? I think our code is not ready for this.

Reply:
How about we have an alias sth like using Loss = Layer?

Reply:
The loss_layer was already opened. However, there was no way to choose the loss type for the created loss layer.
This patch allows that.

Reply:
> How about we have an alias sth like using Loss = Layer?

creating two namespace purpose is to keep loss and layer decoupled. If later, loss is removed a layer, it would require updates here.

Reply:
Okay